src/server/accounting.c

```diff
@@ -1147,56 +1147,60 @@ account_jobend(job *pjob, char *used, int type)
 	len -= i;
 
 	/* finally add on resources used from req_jobobit() */
-	if (((used == NULL) && (pjob->ji_acctrec == NULL) && (type == PBS_ACCT_END)) ||
-		!(used && strstr(used, "resources_used") && (type == PBS_ACCT_RERUN))) {
-		/* If pbs_server is restarted during the end of job processing then used maybe NULL.
-		 * So we try to derive the resource usage information from resources_used attribute of
-		 * the job and then reconstruct the resources usage information into resc_used buffer.
-		 */
-		if (pjob->ji_wattr[(int)JOB_ATR_resc_used].at_user_encoded != NULL)
-			patlist = pjob->ji_wattr[(int)JOB_ATR_resc_used].at_user_encoded;
-		else if (pjob->ji_wattr[(int)JOB_ATR_resc_used].at_priv_encoded != NULL)
-			patlist = pjob->ji_wattr[(int)JOB_ATR_resc_used].at_priv_encoded;
-		else
-			encode_resc(&pjob->ji_wattr[(int)JOB_ATR_resc_used],
-				&temp_head, job_attr_def[(int)JOB_ATR_resc_used].at_name,
-				(char *)0, ATR_ENCODE_CLIENT, &patlist);
+	if (type == PBS_ACCT_END || type == PBS_ACCT_RERUN) {
+		if ((used == NULL && pjob->ji_acctrec == NULL) || (used != NULL && strstr(used, "resources_used") == NULL)) {
+			/* If pbs_server is restarted during the end of job processing then used maybe NULL.
+			 * So we try to derive the resource usage information from resources_used attribute of
+			 * the job and then reconstruct the resources usage information into resc_used buffer.
+			 */
+			if (pjob->ji_wattr[(int) JOB_ATR_resc_used].at_user_encoded != NULL)
+				patlist = pjob->ji_wattr[(int) JOB_ATR_resc_used].at_user_encoded;
+			else if (pjob->ji_wattr[(int) JOB_ATR_resc_used].at_priv_encoded != NULL)
+				patlist = pjob->ji_wattr[(int) JOB_ATR_resc_used].at_priv_encoded;
+			else
+				encode_resc(&pjob->ji_wattr[(int) JOB_ATR_resc_used],
+					    &temp_head, job_attr_def[(int) JOB_ATR_resc_used].at_name,
+					    (char *) 0, ATR_ENCODE_CLIENT, &patlist);
+
+			/* Allocate initial space for resc_used.  Future space will be allocated by pbs_strcat(). */
+			resc_used = malloc(RESC_USED_BUF_SIZE);
+			if (resc_used == NULL)
+				goto writeit;
```

Comment:
I'm not sure I understand your comment.  We're not continuing on, writeit is at the end of the function.  As for going to writeit, I was just following what the rest of the function did on a memory error.  I believe the reason we go to writeit is to write what we have as an accounting record since we can't get the memory to continue it.